From 5da7de3baa22e99ac538f4f81af414f5c3050612 Mon Sep 17 00:00:00 2001
From: Theo GOUREAU <theo.goureau-ext@st.com>
Date: Mon, 21 Oct 2024 10:44:28 +0200
Subject: [PATCH 1333/1695] dts: stm32: remove LTDC rotation from MP21-DK
 RIF/RESMEM configuration

Remove LTDC rotation from MP21-DK RIF configuration: remove its RISUP
and its DDR memory region protected by RISAF4. Remove LTDC rotation
from MP21-DK RESMEM configuration.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git ee793fbc9d334b3c6053156953ea948ef6ff8213 dts: stm32: remove LTDC rotation from MP21-DK RIF/RESMEM configuration]
Change-Id: I5c6a8f9f9e55908feebb3fd126dd271f64f0caf3
Signed-off-by: Theo GOUREAU <theo.goureau-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/412801
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 .../arm/dts/stm32mp215f-dk-ca35tdcid-resmem.dtsi     | 12 +++---------
 core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-rif.dtsi  |  7 +------
 2 files changed, 4 insertions(+), 15 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-resmem.dtsi b/core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-resmem.dtsi
index 404cc1b64..466781236 100644
--- a/core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-resmem.dtsi
+++ b/core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-resmem.dtsi
@@ -105,19 +105,13 @@
 		};
 
 		linuxkernel1: linuxkernel1@84000000 {
-			reg = <0x0 0x84000000 0x0 0x7a800000>;
+			reg = <0x0 0x84000000 0x0 0x7b800000>;
 			no-map;
 		};
 
-		ltdc_sec_layer: ltdc-sec-layer@fe800000 {
-			reg = <0x0 0xfe800000 0x0 0x800000>;
+		ltdc_sec_layer: ltdc-sec-layer@ff800000 {
+			reg = <0x0 0xff800000 0x0 0x800000>;
 			no-map;
 		};
-
-		ltdc_sec_rotation: ltdc-sec-rotation@ff000000 {
-			reg = <0x0 0xff000000 0x0 0x1000000>;
-			no-map;
-		};
-
 	};
 };
diff --git a/core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-rif.dtsi b/core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-rif.dtsi
index 06a48c60e..34faa2acd 100644
--- a/core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-rif.dtsi
+++ b/core/arch/arm/dts/stm32mp215f-dk-ca35tdcid-rif.dtsi
@@ -73,7 +73,6 @@
 		RIFPROT(STM32MP21_RIFSC_LTDC_CMN_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(STM32MP21_RIFSC_LTDC_L0L1_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(STM32MP21_RIFSC_LTDC_L2_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
-		RIFPROT(STM32MP21_RIFSC_LTDC_ROT_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(STM32MP21_RIFSC_CSI_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(STM32MP21_RIFSC_DCMIPP_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
 		RIFPROT(STM32MP21_RIFSC_DCMI_PSSI_ID, RIF_UNUSED, RIF_UNLOCK, RIF_NSEC, RIF_NPRIV, RIF_UNUSED, RIF_SEM_DIS, RIF_CFDIS)
@@ -778,10 +777,6 @@
 	st,protreg = <RISAFPROT(RISAF_REG_ID(13), RIF_CID0_BF|RIF_CID1_BF, RIF_CID0_BF|RIF_CID1_BF, RIF_CID1_BF, RIF_NSEC, RIF_ENC_DIS, RIF_BREN_EN)>;
 };
 
-&ltdc_sec_rotation {
-	st,protreg = <RISAFPROT(RISAF_REG_ID(14), RIF_CID0_BF|RIF_CID1_BF, RIF_CID0_BF|RIF_CID1_BF, RIF_UNUSED, RIF_NSEC, RIF_ENC_DIS, RIF_BREN_EN)>;
-};
-
 &risaf4 {
-	memory-region= <&tfm_code>, <&cm33_cube_fw>, <&tfm_data>, <&cm33_cube_data>, <&ipc_shmem>, <&spare1>, <&bl31_context>, <&op_tee>, <&linuxkernel1>, <&ltdc_sec_layer>, <&ltdc_sec_rotation>;
+	memory-region= <&tfm_code>, <&cm33_cube_fw>, <&tfm_data>, <&cm33_cube_data>, <&ipc_shmem>, <&spare1>, <&bl31_context>, <&op_tee>, <&linuxkernel1>, <&ltdc_sec_layer>;
 };
-- 
2.47.2

